core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1878,6 +1878,98 @@ class SocketServerTest {
     }, false)
   }
 
+  /**
+   * Tests for KAFKA-13559 - Processing request got delayed by 300 ms in the following condition:
+   * 1. Client-Server communication uses SSL socket.
+   * 2. More than one requests are read from the socket into SslTransportLayer.netReadBuffer.
+   *
+   * This 300 ms delay occurs because the socket has no data but the buffer has data. And the sequence of events that
+   * leads to this situation is the following (from the server point of view):
+   *
+   * Step 1 - SslTransportLayer receives more than one requests in the socket and put it in the buffer
+   *          (SslTransportLayer.netReadBuffer).
+   * Step 2 - SslTransportLayer reads all of the bytes and stores it in SslTransportLayer.appReadBuffer.
+   * Step 3 - Process the first request, leaving the second request in SslTransportLayer.appReadBuffer.
+   * Step 4 - THIS IS WHERE THE DELAY IS. Process the second request. This request is read from
+   * SslTransportLayer.appReadBuffer, instead of the socket. Because of this, "select(timeout)" in Selector.poll()
+   * should not block for 300 ms.
+   *
+   * This test is implemented following "makeSocketWithBufferedRequests()" method by putting two requests directly
+   * into SslTransportLayer.netReadBuffer and manually trigger the processing.
```

Comment:
This comment is written largely in the context of the JIRA ticket. I think we can write it more in the context of the current code, eg: replace references to `300 ms` with references to `poll timeout` so that if the poll timeout is changed, it is still consistent.

Reply:
Got it. Changing "300 ms" to "poll timeout".